src/cli/subcommands/f3_cmd.rs

```diff
@@ -119,6 +129,67 @@ impl F3Commands {
             }
             Self::Certs(cmd) => cmd.run(client).await,
             Self::PowerTable(cmd) => cmd.run(client).await,
+            Self::Ready { wait, threshold } => {
+                let is_running = client.call(F3IsRunning::request(())?).await?;
+                if !is_running {
+                    anyhow::bail!("F3 is not running");
+                }
+
+                async fn get_heads(
+                    client: &rpc::Client,
+                ) -> anyhow::Result<(Tipset, FinalityCertificate)> {
+                    let cert_head = client.call(F3GetLatestCertificate::request(())?).await?;
+                    let chain_head = client.call(ChainHead::request(())?).await?;
+                    Ok((chain_head, cert_head))
+                }
+
+                let ticker = Ticker::new(0.., Duration::from_secs(1));
+                let mut stdout = std::io::stdout();
+                let mut text = String::new();
+                for _ in ticker {
+                    match get_heads(&client).await {
+                        Ok((chain_head, cert_head)) => {
+                            if !text.is_empty() {
```

Comment:
We can use `ProgressBar::new_spinner()` here when `wait` flag is set for more intuitive/clear response in the terminal.

Reply:
Good idea, switched to ProgressBar